prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -101,11 +101,20 @@ bool rxr_pkt_req_supported_by_peer(int req_type, struct rdm_peer *peer)
 
 size_t rxr_pkt_req_data_size(struct rxr_pkt_entry *pkt_entry)
 {
-	size_t hdr_size;
+	int pkt_type, read_iov_count;
+	size_t pkt_data_offset;
 
-	hdr_size = rxr_pkt_req_hdr_size(pkt_entry);
-	assert(hdr_size > 0);
-	return pkt_entry->pkt_size - hdr_size;
+	pkt_type = rxr_get_base_hdr(pkt_entry->pkt)->type;
+	pkt_data_offset = rxr_pkt_req_hdr_size(pkt_entry);
+	assert(pkt_data_offset > 0);
+
+	if (pkt_type == RXR_RUNTREAD_MSGRTM_PKT ||
+	    pkt_type == RXR_RUNTREAD_TAGRTM_PKT) {
+		read_iov_count = rxr_get_runtread_rtm_base_hdr(pkt_entry->pkt)->read_iov_count;
+		pkt_data_offset +=  read_iov_count * sizeof(struct fi_rma_iov);
```

Comment:
So each runt-rtm packet will contain pkt_header + read_iov (sender buffer info) + data, right?

Reply:
Yes, `read_iov` will be same. data will be different. most of the header will be same, except `seg_offset`.